java/arcs/core/storage/StorageProxy.kt

```diff
@@ -96,14 +89,17 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperation, T>(
      */
     suspend fun applyOp(op: Op): Boolean {
         log.debug { "Applying operation: $op" }
-        return mutex.withLock {
-            if (!crdt.applyOperation(op)) return@withLock false
+        val localSuccess = syncMutex.withLock { crdt.applyOperation(op) }
+        if (!localSuccess) return false
 
-            val msg = ProxyMessage.Operations<Data, Op, T>(listOf(op), null)
-            store.onProxyMessage(msg)
-            notifyUpdate(op)
-            true
-        }
+        val msg = ProxyMessage.Operations<Data, Op, T>(listOf(op), null)
+        val storeSuccess = store.onProxyMessage(msg)
+
+        if (!storeSuccess) return false
```

Comment:
The old logic here (returning true as long as local crdt change succeeds, regardless of store change) matched the TS logic.

This change makes more sense to me; but it feels like something more needs to happen? Now the local model is out of sync?

Reply:
Agreed, there's something to work through here... I'll leave a TODO.